docs/sdk/v4/guides/advanced/01-pool-data.md

```diff
@@ -0,0 +1,344 @@
+
+This guide will **cover**:
+
+1. Computing the PoolId out of PoolKey
+2. Referencing the StateView contract and fetching metadata
+3. Fetching the positions of all initialized Ticks with multicall
+4. Fetching all ticks by their indices with a multicall
+5. Constructing the Pool object
+
+At the end of the guide, we will have created a `Pool` Object that accurately represents the state of a v4 pool at the time we fetched it.
+
+For this guide, the following Uniswap packages are used:
+  
+- [`@uniswap/v4-sdk`](https://www.npmjs.com/package/@uniswap/v4-sdk)
+- [`@uniswap/sdk-core`](https://www.npmjs.com/package/@uniswap/sdk-core)
+
+We will also use the `ethers-multicall` npm package:
+
+- [`ethers-multicall`](https://www.npmjs.com/package/ethers-multicall)
+
+## Configuration
+We will first create an example configuration `CurrentConfig` in `config.ts`. It has the format:
+
+```typescript
+export const CurrentConfig: ExampleConfig = {
+  env: Environment.MAINNET,
+  rpc: {
+    local: 'http://localhost:8545',
+    mainnet: 'https://mainnet.infura.io/v3/YOUR_API_KEY',
+  },
+  ...
+  poolKey: {
+    currency0: USDC_TOKEN.address,
+    currency1: ETH_TOKEN.address,
+    fee: FEE_AMOUNT_LOW,
+    tickSpacing: TICK_SPACING_TEN,
+    hooks: EMPTY_HOOK,
+  },
+}
+```
+
+The pool used is defined by a pair of tokens in `constants.ts`.
+You can also change these two tokens and the other pool parameters in the config, just make sure a pool actually exists for your configuration.
+Check out the top pools on [Uniswap](https://app.uniswap.org/#/pools).
+
+```typescript
+export const ETH_TOKEN = new Token(
+  SupportedChainId.MAINNET,
+  '0x0000000000000000000000000000000000000000',
+  18,
+  'ETH',
+  'Ether'
+)
+
+export const USDC_TOKEN = new Token(
+  SupportedChainId.MAINNET,
+  '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
+  6,
+  'USDC',
+  'USD//C'
+)
+```
+
+## Computing the PoolId out of PoolKey
+
+In this example, we will construct the **USDC - ETH** Pool with **LOW** fees and without hooks. The SDK provides a method to compute the `PoolId` for this pool:
+
+```typescript
+import { Pool } from '@uniswap/v4-sdk';
+
+const {currency0, currency1, fee, tickSpacing, hooks} = CurrentConfig.poolKey;
+const poolId = Pool.getPoolId(currency0, currency1, fee, tickSpacing, hooks);
+
+```
+
+## Referencing the StateView contract and fetching metadata
+
+Now that we have the `PoolId` of a **USDC - ETH** Pool, we need to call [StateView](https://docs.uniswap.org/contracts/v4/guides/state-view) contract to get the pool state. In v4 you need to use `StateLibrary` to read pool state, but offchain systems—such as frontends or analytics services—require a deployed contract with view functions. This is where `StateView` comes in.
+To construct the Contract we need to provide the address of the contract, its ABI and a provider connected to an [RPC endpoint](https://www.chainnodes.org/docs).
```

Comment:
if were gonna list chainnodes, we should list other RPC providers such as infura, alchemy, llamanodes, etc